dev/environment-dev.yml

```diff
@@ -24,7 +25,7 @@ dependencies:
   # micromamba dependencies
   - cli11 >=2.2
   # micromamba test dependencies
-  - python =3.9  # Some not too recent version
+  - python
```

Comment:
I actually don't know the reason of pinning python here... Maybe this is hiding some incompatibilities. Let's see what happens with the CI.

Reply:
Ok I think unpinning python makes it get `fmt 11` everywhere while #3368 only handles a part of it in the codebase.
I suggest restoring python pinning for this PR and handle `fmt 11` in another one (I will open an issue for it).
Is it something blocking at the moment or more like an enhancement opening for more possibilities?

That being said, that's only showing one issue of possible other ones when unpinning python, so that would be something to check within the fixing PR.